spatialmath/box.go

```diff
@@ -352,3 +355,91 @@ func separatingAxisTest(positionDelta, plane r3.Vector, halfSizeA, halfSizeB [3]
 	}
 	return sum
 }
+
+// ToPointCloud converts a box geometry into a []r3.Vector.
+func (b *box) ToPointCloud(resolution float64) []r3.Vector {
+	// check for user defined spacing
+	var iter float64
+	if resolution != 0. {
+		iter = resolution
+	} else {
+		iter = defaultBoxPointDensity
+	}
+
+	// create points on box faces with box centered at (0, 0, 0)
+	// TODO: add more detailed explanation
```

Comment:
Adding some more detailed comments here is a good idea this loop is a bit complicated.